lib/src/widgets/board.dart

```diff
@@ -375,50 +375,59 @@ class _BoardState extends State<Chessboard> {
 
     final enableListeners = widget.interactive || settings.drawShape.enable;
 
-    final board = Listener(
-      onPointerDown: enableListeners ? _onPointerDown : null,
-      onPointerMove: enableListeners ? _onPointerMove : null,
-      onPointerUp: enableListeners ? _onPointerUp : null,
-      onPointerCancel: enableListeners ? _onPointerCancel : null,
-      child: SizedBox.square(
-        key: const ValueKey('board-container'),
-        dimension: widget.size,
-        child: Stack(
-          alignment: Alignment.topLeft,
-          clipBehavior: Clip.none,
-          children: [
-            if (settings.border == null &&
-                (settings.boxShadow.isNotEmpty ||
-                    settings.borderRadius != BorderRadius.zero))
-              Container(
-                key: const ValueKey('background-container'),
-                clipBehavior: Clip.hardEdge,
-                decoration: BoxDecoration(
-                  borderRadius: settings.borderRadius,
-                  boxShadow: settings.boxShadow,
-                ),
-                child: Stack(
-                  alignment: Alignment.topLeft,
-                  children: highlightedBackground,
+    final board = GestureDetector(
```

Comment:
Oh I forgot that we weren't using a `GestureDetector` when I suggested to use it... 😄 

What you wrote looks fine, but I wonder if there is any drawback to have the `Listener` as a child of a `GestureDetector`?

Reply:
Yeah that was the reason why I also initially hesitated to add it :D but I think with HitTestBehavior.translucent it shouldn't cause any problems, all the events will be passed along to the child, regardless of whether we handled them or not

Reply:
(and the tests still pass, so looks like at least there aren't any regression (can't promise no new bugs of course)

Reply:
My concern was if the events were kind of preempted by GestureDetector, causing unnecessary delays on the Listener. But I don't think it is the case actually (we need to ensure it though).

Reply:
I guess we could also switch them around? First the listener, then the GestureDetector - that way of there is a delay, it won't affect piece movement

Reply:
I think we should not use the `GestureDetector`, better to leave the code as is, since it is a crucial part.

https://github.com/lichess-org/flutter-chessground/blob/bc1bea6903749f2874928dc6d3cce3e83527ce63/lib/src/widgets/board.dart#L738

^ we have already a pointer up event handler and I believe that calling the `onTappedSquare` after this line is all we need to do to support this feature.

Reply:
Done :+1: 

To ensure that the callback is called even if the board is non interactive and shape drawing is disabled (which would be the case for the coordinate trainer), I had to remove the condition that use to disable the _onPointerUp callback in that case